src/tensors/truncation.jl

```diff
@@ -166,21 +168,24 @@ end
 function _truncate!(V::SectorVectorDict, trunc::TruncationCutoff, p = 2)
     G = keytype(V)
     S = real(eltype(valtype(V)))
-    truncdim = SectorDict{G,Int}(c=>length(v) for (c,v) in V)
-    next_val = SectorDict{G,S  }(c=>zero(S)   for (c,v) in V)
+    truncdim = SectorDict{G,Int      }(c=>length(v) for (c,v) in V)
+    next_val = SectorDict{G,Array{S} }(c=>[zero(S)] for (c,v) in V)
     for (c,v) in V
         newdim = findlast(Base.Fix2(>, trunc.ϵ), v)
         if newdim == nothing
             truncdim[c] = 0
-	    next_val[c] = v[1]
-	elseif newdim < truncdim[c]
- 	    truncdim[c] = newdim
-	    next_val[c] = v[newdim+1]
+	    next_val[c] = v
 	else
 	    truncdim[c] = newdim
+            next_val[c] = v[newdim+1:end]
         end
     end
-    trunc.addone && (truncdim[findmax(next_val)[2]]+=1)
+    for i in 1:trunc.add_back
+        key_max = findmax(next_val)[2] 
```

Comment:
Did you test this? Is this not looking for the maximum of a bunch of vectors, which I think is not defined?

Reply:
There is also `argmax` as alternative to the second argument of `findmax`

Reply:
Yes I tested this. 
Also expected it to fail but turns out that it it searching for the vector that contains the largest element.

Reply:
Ok, it indeed searches for the one which is largest according to a semantic order. So this works because the singular values are sorted in descending order. Nice. That makes it very easy indeed.